typed-racket-lib/typed-racket/base-env/base-env.rkt

```diff
@@ -905,51 +905,65 @@
-                            (a b . -> . (-HT a b))
-                            (a b a b . -> . (-HT a b))
-                            (a b a b a b . -> . (-HT a b))
-                            (a b a b a b a b . -> . (-HT a b))))]
-[make-hash (-poly (a b) (->opt [(-lst (-pair a b))] (-HT a b)))]
-[make-hasheq (-poly (a b) (->opt [(-lst (-pair a b))] (-HT a b)))]
-[make-hasheqv (-poly (a b) (->opt [(-lst (-pair a b))] (-HT a b)))]
-[make-weak-hash (-poly (a b) (->opt [(-lst (-pair a b))] (-HT a b)))]
-[make-weak-hasheq (-poly (a b) (->opt [(-lst (-pair a b))] (-HT a b)))]
-[make-weak-hasheqv (-poly (a b) (->opt [(-lst (-pair a b))] (-HT a b)))]
-[make-immutable-hash (-poly (a b) (->opt [(-lst (-pair a b))] (-HT a b)))]
-[make-immutable-hasheq (-poly (a b) (->opt [(-lst (-pair a b))] (-HT a b)))]
-[make-immutable-hasheqv (-poly (a b) (->opt [(-lst (-pair a b))] (-HT a b)))]
-
-[hash-set (-poly (a b) ((-HT a b) a b . -> . (-HT a b)))]
-[hash-set! (-poly (a b) ((-HT a b) a b . -> . -Void))]
+[hash (-poly (a b) (cl->* (-> (-Immutable-HT a b))
+                          (a b . -> . (-Immutable-HT a b))
+                          (a b a b . -> . (-Immutable-HT a b))
+                          (a b a b a b . -> . (-Immutable-HT a b))
+                          (a b a b a b a b . -> . (-Immutable-HT a b))))]
+[hasheqv (-poly (a b) (cl->* (-> (-Immutable-HT a b))
+                             (a b . -> . (-Immutable-HT a b))
+                             (a b a b . -> . (-Immutable-HT a b))
+                             (a b a b a b . -> . (-Immutable-HT a b))
+                             (a b a b a b a b . -> . (-Immutable-HT a b))))]
+[hasheq (-poly (a b) (cl->* (-> (-Immutable-HT a b))
+                            (a b . -> . (-Immutable-HT a b))
+                            (a b a b . -> . (-Immutable-HT a b))
+                            (a b a b a b . -> . (-Immutable-HT a b))
+                            (a b a b a b a b . -> . (-Immutable-HT a b))))]
+[make-hash (-poly (a b) (->opt [(-lst (-pair a b))] (-Mutable-HT a b)))]
+[make-hasheq (-poly (a b) (->opt [(-lst (-pair a b))] (-Mutable-HT a b)))]
+[make-hasheqv (-poly (a b) (->opt [(-lst (-pair a b))] (-Mutable-HT a b)))]
+[make-weak-hash (-poly (a b) (->opt [(-lst (-pair a b))] (-Weak-HT a b)))]
+[make-weak-hasheq (-poly (a b) (->opt [(-lst (-pair a b))] (-Weak-HT a b)))]
+[make-weak-hasheqv (-poly (a b) (->opt [(-lst (-pair a b))] (-Weak-HT a b)))]
+[make-immutable-hash (-poly (a b) (->opt [(-lst (-pair a b))] (-Immutable-HT a b)))]
+[make-immutable-hasheq (-poly (a b) (->opt [(-lst (-pair a b))] (-Immutable-HT a b)))]
+[make-immutable-hasheqv (-poly (a b) (->opt [(-lst (-pair a b))] (-Immutable-HT a b)))]
+
+[hash-set (-poly (a b) (cl-> [((-HT a b) a b) (-Immutable-HT a b)]
+                             [((-Immutable-HT a b) a b) (-Immutable-HT a b)]))]
+[hash-set! (-poly (a b) (cl-> [((-HT a b) a b) -Void]
+                              [((-Mutable-HT a b) a b) -Void]
```

Comment:
No they're not, I'll remove them. But is there any hope we could deprecate the `-HT` signatures?